x-pack/plugins/apm/server/routes/suggestions/get_suggestions_with_terms_aggregation.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { rangeQuery, termQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsWithTermsAggregation({
+  fieldName,
+  fieldValue,
+  serviceName,
+  setup,
+  size,
+  start = moment().subtract(24, 'h').valueOf(),
```

Comment:
I'd prefer for this to be made more explicit - making the route query params required and having the consumer specify them. I think setting default values in the route makes it easy to forget about it.

Reply:
updated https://github.com/elastic/kibana/pull/126679/commits/30465933cd2bddbf6a26abe0eb9386fbd44985df